### ISSUE TITLE
Expose LSP-only diagnostics via reproducible CLI lever

### ISSUE THREAD

**Issue by blightbow** [astral-sh/ty#4385]
## Feature Request

ty reports `Code is unreachable` and `` `x` is unused `` only over the language server, as hints tagged `DiagnosticTag.Unnecessary`. There is no way to obtain the same set from `ty check`, even with `--error all`. The rules have no CLI names, so they cannot be enabled individually either.

For a human in an editor that is the right design, and #784 chose it deliberately. For an LLM coding agent it removes the only verification channel the agent actually has, because an agent's feedback loop is *run a command, read the exit code*. I would like an opt-in, non-failing CLI lever that reproduces the language server's diagnostic set on demand.

## Reproduction

```python
def work() -> None: ...


def cleanup_flag() -> None:
    done = False
    try:
        work()
        done = True
    finally:
        if not done:
            print("cleanup")
```

ty 0.0.74:

```
$ ty check repro.py
All checks passed!

$ ty check --error all repro.py
All checks passed!
```

Same file, same version, over the language server (`ty server`, `textDocument/didOpen`, collecting `publishDiagnostics`):

```
L5   [hint] tag=Unnecessary  `done` is unused
L11  [hint] tag=Unnecessary  Code is unreachable
```

`--error all` is the maximal CLI surface and does not include them. `--error unused-parameter` returns `warning[unknown-rule]: Unknown rule`.

## This is particularly bad for LLMs.

An agent verifies its work by running commands, and I use both pytest-ruff and pytest-ty as additional quality gates. Anything reachable that way gets fixed, because the agent can prove to itself that the diagnostics are valid and solvable. The problem lies in the fact that the LLM cannot reproduce the results of the LSP server. Once its output starts to deviate from the output of the `ty` CLI (particularly when a coding harness accumulates stale results), the LLM begins to confabulate explanations for the difference. In my experience, the most common confabulation is stale caching in a text editor the user has open. The LSP (let alone the coding harness integrating with the LSP) does not prime the LLM with sufficient information in the output for it to understand the failure mode, and there is no lever to validate the results. Net result: output from the LSP is treated as noise to be ignored, which is very bad when paired with diagnostic feedback that cannot be produced outside of the ignored LSP.

Claude Code is the headline case, because its LSP implementation has a staleness bug causing diagnostic messages to accumulate over Read/Edit tool calls with no pressure release for when those files are modified outside of those tools. ([anthropics/claude-code 76870](https://github.com/anthropics/claude-code/issues/76870))

**Comment by MichaReiser** [comment]
Thanks, this is https://github.com/astral-sh/ty/issues/4033
